client/runtime/src/main/java/io/quarkiverse/openapi/generator/providers/ConfigCredentialsProvider.java

```diff
@@ -63,4 +64,8 @@ public String getBearerToken(ClientRequestContext requestContext, String openApi
                 .orElse("");
     }
 
+    @Override
+    public String getOauth2BearerToken(ClientRequestContext requestContext, String openApiSpecId, String authName) {
```

Comment:
You're only using the `requestContext` parameter here.

Reply:
Yes but I wanted to keep the same signature as the other methods declared in the interface, in case the user, for they own implementation, need to access specific properties.
I can completely remove it and users can open PR to add new arguments if needed

Reply:
Yes, I think that's more doable. We can overload the interface if needed in the future.

Reply:
@ricardozanini Now that I am implementing the oauth2 token exchange pattern I see that I need those `openApiSpecId` and `authName` to only perform the oauth2 token exchange if the user enabled it for the given spec and security scheme; I do not want to exchange the token for all OAUTH2 security scheme

Now, if I extend the interface to add this new method `String getOauth2BearerToken(ClientRequestContext requestContext, String openApiSpecId, String authName)` and that my custom class extends `ConfigCredentialsProvider` and implements the newly created interface, the issue lies in the invocation of this new method as the interface implementation is called by the providers and they call `String getOauth2BearerToken(ClientRequestContext requestContext)`

How do you want to proceed?